internal/transform/scope-css_test.go

```diff
@@ -239,7 +239,36 @@ func TestScopeStyle(t *testing.T) {
   color: blue
   font-size: 18px;
 }`,
-			want: `.foo.astro-XXXXXX{color:blue font-size:18px;}`,
+			want: `.foo.astro-XXXXXX{color:blue font-size: 18px}`,
+		},
+		{
+			name:   "nesting media",
+			source: ":global(html) { @media (min-width: 640px) { color: blue } }html { background-color: lime }",
+			want:   "html{@media (min-width: 640px){color:blue}}html{background-color:lime}",
+		},
+		{
+			name:   "nesting combinator",
+			source: "div { & span { color: blue } }",
+			want:   "div.astro-XXXXXX{& span.astro-XXXXXX{color:blue}}",
+		},
+		{
+			name:   "nesting modifier",
+			source: ".header { background-color: white; &.dark { background-color: blue; }}",
+			want:   ".header.astro-XXXXXX{background-color:white;&.dark{background-color:blue}}",
+		},
+		{
+			name: "@container",
+			source: `@container (min-width: 200px) and (min-height: 200px) {
+        h1 {
+          font-size: 30px;
+        }
+      }`,
+			want: "@container (min-width: 200px) and (min-height: 200px){h1.astro-XXXXXX{font-size:30px}}",
+		},
+		{
+			name:   "@layer",
+			source: "@layer theme, layout, utilities; @layer special { .item { color: rebeccapurple; }}",
+			want:   "@layer theme,layout,utilities;@layer special{.item.astro-XXXXXX{color:rebeccapurple}}",
```

Comment:
New tests for modern CSS `draft` features. There's no toggle or user setting, we just know how to parse it now.